### ISSUE TITLE
Stoping Picroft to narrate last query

### ISSUE THREAD

**Issue by rajdpandey** [MycroftAI/mycroft-skills#1023]
Hi all,

I am using picroft on Raspberry Pi 3B+ and there is an issue I wanted to ask about and how to achieve that.

Example:

User: Hey Mycroft tell me about the Himalayas?
Mycroft: Starts narrating answer.
(Between the Narration)
User asks: Hey Mycroft tell me about America?

Then Mycroft should stop tell me about Himalaya's and should tell me about America.

But right now it first completes whole Himalaya's narration and then starts telling me about America.

I don't want to manually say Hey Mycroft stop.

It should be automatic.

So how can I make this?

**Comment by forslund** [comment]
Hi @rajdpandey!

For a general solution a mycroft-core change where speech is stopped when a new utterance has been processed would be needed. A specific solution for the wiki would be modifying the [wiki skill](https://github.com/MycroftAI/skill-wiki). Making it send a stop signal when starting to handle a new "tell me about..." intent.

I suggest moving this issue either to mycroft-core or the wiki-skill.

**Comment by rajdpandey** [comment]
Ok but what about If I am asking to Play Song and In between tell it to tell me about something? So after telling it would again start playing the song.

So I am asking that shouldn't this be an automatic feature that on new utterance it should stop the last one.

I will also move this to mycroft-core and wiki-skill.

**Comment by forslund** [comment]
The song thing is partially implemented, it will pause music for a short while when asked a question. This something currently being expanded upon since previously it was very hard to get voice commands in during playback of other things

**Comment by rajdpandey** [comment]
Yeah so for news and wiki also. So can you guide me to the documentation of function used in creating skills so I can look up and find something to edit wiki-skill and make work as wanted?

**Comment by krisgesling** [comment]
If you're only looking to do it on your own versions of skills then you can take a look at how the [Stop Skill](https://github.com/MycroftAI/skill-stop/blob/19.02/__init__.py) implements this.

**Comment by rajdpandey** [comment]
I am thinking that it should be applied to main also but right now I am gonna test it on my skill.
